### PR TITLE
Mousa

### PR DESCRIPTION
**Pull request by mosessaleh**

### CODE REVIEW

**Comment by jacobandresen** [comment]
looks like you added some extra stuff into the category route? Maybe you could benefit from having a content table instead?

File: server/knexfile.js [hunk 1]
```diff
@@ -1,28 +1,29 @@
...
+      user:     'root',
+      password: ''
     },
     pool: {
       min: 2,
       max: 10
     },
     migrations: {
-      tableName: 'knex_migrations'
+      tableName: 'categories'
```
Comment (jacobandresen) [inline server/knexfile.js]: this messes up knex. The knex_migrations table is used by knex to check which migrations has been run.

File: client/src/components/AddForm.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        event.preventDefault()
+        var data = {
+            category: this.state.category,
+            difficulty: this.state.difficulty,
+            link: this.state.link,
+            title: this.state.title,
+            type: this.state.type,
+            description:this.state.description
+        }
+        fetch('/newContent', {
```
Comment (jacobandresen) [inline client/src/components/AddForm.js]: You are submitting to the newContent route here. It currently inserts into the Categories table.  How about a Contents table instead? .. and content route?

File: client/src/components/CategoryView.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                                <br /><br />
+                                <p>Caregory: <b>{res.category}</b></p>
+                                <p>Type: <b>{res.type}</b></p>
+                                <p>Difficulty: <b>{res.difficulty}</b></p>
+                                <b>Description:</b><br />
+                                <p>{res.description}</p>
+                                <p>You can visit <a target='_blanck' href={res.link}>this</a> to get more information about this content</p>
+                                <br /><br /><br /><br />
+                                <button onClick={this.likeIt} >Like it ({res.voteUp})</button>{' '}
+                                <button onClick={this.unLikeIt} >Unlike it ({res.voteDown})</button>
```
Comment (jacobandresen) [inline client/src/components/CategoryView.js]: I don't remember seeing unlike as a requirement ?

File: client/src/components/AddForm.js (hunk 2 above)
Comment (mosessaleh) [reply client/src/components/AddForm.js]: Yes it is, I just made a mistake in naming.

File: client/src/components/CategoryView.js (hunk 3 above)
Comment (mosessaleh) [reply client/src/components/CategoryView.js]: it is like a vote, I can change it to 'vote' name

Comment (jacobandresen) [reply client/src/components/CategoryView.js]: hehe. nevermind. Did you discuss this with your team? Looks like a cool feature. I wonder if zaki will misuse this during your demo :P

Comment (mosessaleh) [reply client/src/components/CategoryView.js]: I think that Ismail has checked this. We talked about it and he has the link to my request

Comment (jacobandresen) [reply client/src/components/CategoryView.js]: cool.

**Pull request merged**